src/gui/torrentoptionsdialog.ui

```diff
@@ -6,14 +6,44 @@
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>390</width>
-    <height>450</height>
+    <width>450</width>
+    <height>506</height>
    </rect>
   </property>
   <property name="windowTitle">
    <string>Torrent Options</string>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_2">
+   <item>
+    <widget class="QCheckBox" name="checkAutoTMM">
+     <property name="toolTip">
+      <string>Automatic mode means that various torrent properties(e.g. save path) will be decided by the associated category</string>
+     </property>
+     <property name="text">
+      <string>Automatic Torrent Management</string>
+     </property>
+    </widget>
+   </item>
+   <item>
+    <layout class="QHBoxLayout" name="horizontalLayout">
+     <item>
+      <widget class="QLabel" name="labelSavePath">
+       <property name="sizePolicy">
+        <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
+         <horstretch>0</horstretch>
+         <verstretch>0</verstretch>
+        </sizepolicy>
+       </property>
+       <property name="text">
+        <string>Save path:</string>
+       </property>
+      </widget>
+     </item>
+     <item>
+      <widget class="FileSystemPathLineEdit" name="savePath" native="true"/>
```

Comment:
And set "Horizontal policy" to "Expanding".

Reply:
All comments addressed so far.